plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -591,24 +666,37 @@ function OPDSBrowser:genItemTableFromCatalog(catalog, item_url)
         item.content = entry.content or entry.summary
         table.insert(item_table, item)
     end
+
+    if next(self.facet_groups) == nil then self.facet_groups = nil end -- Clear if empty
+
     return item_table
 end
 
 -- Requests and shows updated list of catalog entries
 function OPDSBrowser:updateCatalog(item_url, paths_updated)
     local menu_table = self:genItemTableFromURL(item_url)
-    if #menu_table > 0 then
+    if #menu_table > 0 or self.facet_groups then
         if not paths_updated then
             table.insert(self.paths, {
                 url   = item_url,
                 title = self.catalog_title,
             })
         end
         self:switchItemTable(self.catalog_title, menu_table)
-        self:setTitleBarLeftIcon("plus")
-        self.onLeftButtonTap = function()
-            self:addSubCatalog(item_url)
+
+        -- Set appropriate title bar icon based on content
+        if self.facet_groups or self.search_url then
+            self:setTitleBarLeftIcon("appbar.settings")
```

Comment:
`appbar.menu`

I like your idea to move the Search item from the list to the menu.